backend/routes_announcements.py

```
@@ -0,0 +1,332 @@
+            return jsonify({'error': f'An error occurred: {str(e)}'}), 500
+
+    # Mark an announcement as read
+    @app.route('/api/announcements/<int:id>/read', methods=['POST'])
+    def mark_announcement_read(id):
+        try:
+            # Check if user is logged in
+            if 'user_id' not in session:
+                return jsonify({'error': 'Authentication required'}), 401
+
+            # Get the announcement
+            announcement = Announcement.query.get_or_404(id)
```

Ruff (0.11.9)

Local variable `announcement` is assigned to but never used

Remove assignment to unused variable `announcement`

(F841)